lib/puppet/provider/x509_cert/openssl.rb

```diff
@@ -22,11 +22,39 @@ def self.check_private_key(resource)
     cert.check_private_key(priv)
   end
 
+  def self.old_cert_is_equal(resource)
+    cert = OpenSSL::X509::Certificate.new(File.read(resource[:path]))
+
+    altName = ''
+    cert.extensions.each do |ext|
+        altName = ext.value if ext.oid == 'subjectAltName'
+    end
+
+    subjectName = ''
+    cert.subject.to_s.split('/').each do |name|
+        k,v = name.split('=')
+          subjectName = v if k == 'CN'
+    end
+
+    require File.expand_path('../../../../../../inifile/lib/puppet/util/ini_file', __FILE__)
```

Comment:
Yeah, I know. Also tests should be touched a little. To every thing there is a season, you know :)